bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareUpdateService.java

```diff
@@ -244,6 +244,43 @@ public Void call() {
         });
     }
 
+    /**
+     * Cancels the firmware update of the thing having the given thing UID by invoking the operation
+     * {@link FirmwareUpdateHandler#cancel()} of the thing´s firmware update handler.
+     * 
+     * @param thingUID the thing UID (must not be null)
+     */
+    public void cancelFirmwareUpdate(final ThingUID thingUID) {
+        Preconditions.checkNotNull(thingUID, "Thing UID must not be null.");
+        final FirmwareUpdateHandler firmwareUpdateHandler = getFirmwareUpdateHandler(thingUID);
+        if (firmwareUpdateHandler == null) {
+            throw new IllegalArgumentException(
+                    String.format("There is no firmware update handler for thing with UID %s.", thingUID));
+        }
+        getPool().submit(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    SafeMethodCaller.call(new SafeMethodCaller.ActionWithException<Void>() {
+                        @Override
+                        public Void call() {
+                            logger.debug("Canceling firmware update for thing with UID {}.", thingUID);
+                            firmwareUpdateHandler.cancel();
```

Comment:
How is the caller notified that the cancellation of the FW udpate was successful?